core/src/context.rs

```diff
@@ -276,6 +274,20 @@ impl Context {
         }
     }
 
+    pub fn verify_signature(
```

Comment:
Why is this function in `Context`? It doesn't even use `self`. I think these three lines should go straight into `invoke_verify_signatureinvoke_verify_signature` instead of calling this function on `context`.

For context ;), `Context::sign` is in here because it wraps the conductor API callback in way that code deeper in core does not need to handle JSONRPC. But verification can happen locally, so I don't see why we would want to put it in `Context`.

Reply:
I agree with this in principle, but I think it will just get handled if we do what I asked below of having the arg be a provenance, because verify is defined on a provenance.